### PR TITLE
comment bar and campaign page design rehaul connect to #309

### PR DESCRIPTION
**Pull request by hoop71**
Pretty big PR here. Integrating from changes from @TheBrooks as well as style and logic for the comments, Campaign.js re-design, and Sig Bar logic and style edits. 

Cheers, 

bitmoji

### CODE REVIEW

File: backend/routes/wasteProviderRoutes.js [hunk 1]
```diff
@@ -57,4 +57,4 @@ module.exports = app => {
     });
   });
   */
-};
+};
```
Comment (TheBrooks) [inline backend/routes/wasteProviderRoutes.js]: if this is all that changed,  revert the changes in this file

File: client/src/components/Comment.js [hunk 2]
```diff
@@ -15,28 +15,42 @@ class Comment extends Component {
 
   findTime() {
     const seconds = Math.floor((new Date() - new Date(this.props.dateAdded)) / 1000);
-    let interval = Math.floor(seconds / 31536000);
-
-    if (interval > 1) {
+    let interval = Math.floor(seconds / 31536000); // one year in seconds
```
Comment (TheBrooks) [inline client/src/components/Comment.js]: no magic numbers. please pull this out into a constant
const ONE_YEAR_IN_SECONDS = 365 * 24 * 60 * 60;

File: client/src/components/Comment.js [hunk 3]
```diff
@@ -15,28 +15,42 @@ class Comment extends Component {
...
+    if (interval >= 1) {
+      if (interval === 1) {
+        return `${interval} year`;
+      }
       return `${interval} years`;
     }
     interval = Math.floor(seconds / 2592000);
```
Comment (TheBrooks) [inline client/src/components/Comment.js]: same as above
const ONE_MONTH_IN_SECONDS = 30 * 24 * 60 * 60;

File: client/src/components/Comment.js [hunk 4]
```diff
@@ -15,28 +15,42 @@ class Comment extends Component {
...
     }
     interval = Math.floor(seconds / 3600);
-    if (interval > 1) {
+    if (interval >= 1) {
+      if (interval === 1) {
+        return `${interval} hour`;
+      }
       return `${interval} hours`;
     }
     interval = Math.floor(seconds / 60);
```
Comment (TheBrooks) [inline client/src/components/Comment.js]: same as above

File: client/src/components/Comment.js [hunk 5]
```diff
@@ -15,28 +15,42 @@ class Comment extends Component {
...
     }
     interval = Math.floor(seconds / 86400);
-    if (interval > 1) {
+    if (interval >= 1) {
+      if (interval === 1) {
+        return `${interval} day`;
+      }
       return `${interval} days`;
     }
     interval = Math.floor(seconds / 3600);
```
Comment (TheBrooks) [inline client/src/components/Comment.js]: same as above

File: client/src/components/Comment.js [hunk 6]
```diff
@@ -15,28 +15,42 @@ class Comment extends Component {
...
     }
     interval = Math.floor(seconds / 2592000);
-    if (interval > 1) {
+    if (interval >= 1) {
+      if (interval === 1) {
+        return `${interval} month`;
+      }
       return `${interval} months`;
     }
     interval = Math.floor(seconds / 86400);
```
Comment (TheBrooks) [inline client/src/components/Comment.js]: same as above

Comment (hoop71) [reply client/src/components/Comment.js]: OK - these were legacy but I've updated them 👍

File: client/src/components/Comment.js [hunk 7]
```diff
@@ -71,8 +85,10 @@ class Comment extends Component {
   render() {
     return (
       <div className="comment">
-        <span style={{ fontWeight: 'bold' }}>{this.props.userName}</span>
-        <span style={{ fontSize: '12px' }}>&nbsp;&bull;&nbsp;{this.findTime()}</span>
+        <span className="comment-name" style={{ fontWeight: 'bold' }}>
```
Comment (TheBrooks) [inline client/src/components/Comment.js]: comment-name class referenced but `.comment-name {}` SCSS class occurs nowhere in the codebase.

Pull `fontWeight` out the style attribute of the span into a `comment-name {}` SCSS class.
Its best to keep all static styling out of the jsx

File: client/src/components/Discussion.js [hunk 8]
```diff
@@ -74,8 +74,8 @@ class Discussion extends Component {
     return (
       <div className="discussion">
         <div className="discussion-header">COMMENTS</div>
-        {this.renderMainPostBox()}
-        {this.renderComments()}
+        <div className="comment-box">{this.renderComments()}</div>
+        <div>{this.renderMainPostBox()}</div>
```
Comment (TheBrooks) [inline client/src/components/Discussion.js]: This flips the comment entry from above the already posted comments to the below.  If there is a big discussion the user would have to scroll down all the way to comment. 
This doesn't have to be changed back but I wanted to discuss the user flow change.

Is this a change we want?

File: client/src/components/MapCard.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+                    bottom: 0,
+                    justifyContent: 'flex-end',
+                    alignItems: 'center'
+                  }}
+                />
+              }
+              mapElement={<div style={{ height: '100%' }} />}
+              onMapLoad={_.noop}
+              onMapClick={_.noop}
+              markers={apartments}
```
Comment (TheBrooks) [inline client/src/components/MapCard.js]: This class looks great.  One small mention however just to have what we talked about in person in writing for a future reminder.

We discussed for MVP changing the campaign map to only have the single marker and also not allowing users to change campaigns by clicking on other markers.  
In the future it may be good to have these ability be settable via props passed in (similar to CollapsePanel prop defaultExpanded).

If you think its an easy change feel free to include it in this PR, if not go ahead and punt it to a new issue and tag it MVP,ready.

File: client/src/components/Tools.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
+import React, { Component } from 'react';
```
Comment (TheBrooks) [inline client/src/components/Tools.js]: Is this class a hard coded version of CollapsePanel?
Could you remove this class and add more passable props to CollapsePanel to achieve the same thing?

File: client/src/components/Comment.js (hunk 7 above)
Comment (hoop71) [reply client/src/components/Comment.js]: 👍

File: client/src/components/Discussion.js (hunk 8 above)
Comment (hoop71) [reply client/src/components/Discussion.js]: Will defer to the group here... I can make a case for both. We should also consider rendering the comments from more recent > less recent instead of vice verse (how it is now)

File: client/src/components/Tools.js (hunk 10 above)
Comment (hoop71) [reply client/src/components/Tools.js]: Yes, it's hard coded at the moment for the ToolList. The class exists here to manage the expanded state of the panel and chose to put it here since it's managing its own state. I am hesitant to store the state of the panel in a different component but happy to talk about wats to improve this.

File: client/src/components/MapCard.js (hunk 9 above)
Comment (hoop71) [reply client/src/components/MapCard.js]: Let's punt this. Should be hard but this PR is already sprawling.

Comment (TheBrooks) [reply client/src/components/MapCard.js]: cool. link the issue created here.

File: client/src/components/Discussion.js (hunk 8 above)
Comment (TheBrooks) [reply client/src/components/Discussion.js]: lets bring this up tonight at the meetup.

File: client/src/components/MapCard.js (hunk 9 above)
Comment (hoop71) [reply client/src/components/MapCard.js]: https://github.com/codefordenver/Circular/issues/325

File: client/src/stylesheets/base/_colors.scss [hunk 11]
```diff
@@ -9,10 +9,13 @@ $secondary-color: #214552;
...
 $icon-row-color: #e3e3e3;
 $main-blue: #176682;
-$sign-out-button: #ff3838;
+$sign-out-button: #c73232;
 
 //comments
 $light-gray: #dbdfe4;
+$comment-gray: #888888;
+$comment-black: #000;
+$comment-box-blue: cornflowerblue;
```
Comment (TheBrooks) [inline client/src/stylesheets/base/_colors.scss]: 'It must be Tuesday... He had on his cornflower blue tie on.'

File: client/src/stylesheets/components/_CampaignPage.scss [hunk 12]
```diff
@@ -1,42 +1,48 @@
...
+  .campaign-page-name {
+    font-size: 1.5em;
+    color: white;
+    margin-left: 1.5rem;
+  }
 
-row {
-  margin-left: 0;
-  margin-right: 0;
-}
```
Comment (TheBrooks) [inline client/src/stylesheets/components/_CampaignPage.scss]: good good

File: client/src/stylesheets/components/_CampaignPage.scss [hunk 13]
```diff
@@ -193,22 +173,11 @@ row {
...
-      font-size: 8px;
-    }
-  }
-
   .status {
     margin-top: 1.5em;
   }
 
-  .status-bar {
+  .status-bar:not(:first-child) {
```
Comment (TheBrooks) [inline client/src/stylesheets/components/_CampaignPage.scss]: what is this :not(:first-child) accomplishing?  
Perhaps include a quick comment?

File: client/src/stylesheets/components/_Card.scss [hunk 14]
```diff
@@ -0,0 +1,43 @@
+$card-border-round: 5px;
+
+.card {
+    border-radius: $card-border-round;
+    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.05), 0 0 0 1px rgba(63, 63, 68, 0.1);
+    background-color: #ffffff;
+    margin-bottom: 30px;
+
+    .card-map {
+        paddding: 1em;
```
Comment (TheBrooks) [inline client/src/stylesheets/components/_Card.scss]: typeo on `padding`

File: client/src/stylesheets/components/_Comment.scss [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+  border-top: 1px solid $comment-box-blue;
+  margin-top: 2em;
+  border-top-left-radius: 10px;
+  -webkit-box-shadow: -1px -1px -1px 0px $comment-box-blue;
+  -moz-box-shadow: -1px -1px -1px 0px $comment-box-blue;
+  box-shadow: -1px -1px -1px 0px $comment-box-blue;
+  text-align: left;
+
+  .comment-name {
+    fontweight: 'bold';
```
Comment (TheBrooks) [inline client/src/stylesheets/components/_Comment.scss]: font-weight

File: client/src/stylesheets/components/_Comment.scss [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  .comment-time {
+    color: $comment-gray;
+    fontsize: 12px;
```
Comment (TheBrooks) [inline client/src/stylesheets/components/_Comment.scss]: font-size

File: client/src/stylesheets/components/_Comment.scss [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  .comment-box {
+    overflow-y: scroll;
+    max-height: 200px;
+  }
+
+  // @media #{$information-phone} {
+  //   padding: 8px 3px 0 6px;
+  // }
```
Comment (TheBrooks) [inline client/src/stylesheets/components/_Comment.scss]: can this comment be deleted?

File: client/src/stylesheets/components/_Tools.scss [hunk 18]
```diff
@@ -0,0 +1,57 @@
+.tool-panel-wrapper {
```
Comment (TheBrooks) [inline client/src/stylesheets/components/_Tools.scss]: this might need to change depending on what happens to the ToolsList component

File: client/src/components/Tools.js [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+import { Link } from 'react-router';
+import { Panel } from 'react-bootstrap';
+
+class ToolList extends Component {
```
Comment (TheBrooks) [inline client/src/components/Tools.js]: Component name and file name should match

File: client/src/stylesheets/components/_collapsePanel.scss [hunk 20]
```diff
@@ -9,9 +9,9 @@
 
   .collpase-panel-title {
```
Comment (TheBrooks) [inline client/src/stylesheets/components/_collapsePanel.scss]: typo: .collpase-panel-title -> .collapse-panel-title

Comment (TheBrooks) [reply client/src/stylesheets/components/_collapsePanel.scss]: this scss typeo class exists in CollapsePanel.js too.  can you swing by and fit it real fast too? Thanks

File: client/src/stylesheets/components/_Tools.scss [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+    font-size: 20px;
+
+    a {
+      color: white;
+      text-decoration: none;
+    }
+  }
+}
+
+@media (max-width: 992px) {
```
Comment (TheBrooks) [inline client/src/stylesheets/components/_Tools.scss]: what size screen is this 992px?  pull this into _global.scss and give it an identifying label

$tablet-phone: 'only screen and (max-width : 992px)';

File: client/src/components/Tools.js (hunk 10 above)
Comment (TheBrooks) [reply client/src/components/Tools.js]: Changed this to a simple named CollapsePanel wrapper.  If desired it can be pulled into the SignCampaign file sense that is the only place where it is used.

File: client/src/components/MapCard.js (hunk 9 above)
Comment (TheBrooks) [reply client/src/components/MapCard.js]: awesome thanks!

File: client/src/stylesheets/components/_Tools.scss (hunk 18 above)
Comment (TheBrooks) [reply client/src/stylesheets/components/_Tools.scss]: This has been renamed to ToolList.scss.

The file now has only the scss needed to style the CollpasePanel body in the ToolList jsx

File: client/src/components/Tools.js (hunk 19 above)
Comment (TheBrooks) [reply client/src/components/Tools.js]: File name changed to ToolList

File: client/src/stylesheets/components/_collapsePanel.scss (hunk 20 above)
Comment (TheBrooks) [reply client/src/stylesheets/components/_collapsePanel.scss]: fixed in all places this typo has occured

File: client/src/stylesheets/components/_Tools.scss (hunk 21 above)
Comment (TheBrooks) [reply client/src/stylesheets/components/_Tools.scss]: Pulled this media query into the CollapsePanel scss but did not make the constant.

File: client/src/stylesheets/components/_ToolList.scss [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+  ul {
+    list-style-type: square;
+    margin-left: 1em;
+    color: $main-blue;
+  }
+  li {
+    text-align: left;
+    font-size: 88%;
+
+    @media (max-width: 992px) {
```
Comment (TheBrooks) [inline client/src/stylesheets/components/_ToolList.scss]: what size screen is this 992px? pull this into _global.scss and give it an identifying label

$tablet-phone: 'only screen and (max-width : 992px)';

File: client/src/stylesheets/components/_CampaignPage.scss (hunk 13 above)
Comment (hoop71) [reply client/src/stylesheets/components/_CampaignPage.scss]: Looks like obsolete code now. It was preventing duplicate margin-top spacing.

File: client/src/stylesheets/components/_Card.scss (hunk 14 above)
Comment (hoop71) [reply client/src/stylesheets/components/_Card.scss]: 👍

File: client/src/stylesheets/components/_Comment.scss (hunk 15 above)
Comment (hoop71) [reply client/src/stylesheets/components/_Comment.scss]: updated and removed "" from bold also

File: client/src/stylesheets/components/_Comment.scss (hunk 16 above)
Comment (hoop71) [reply client/src/stylesheets/components/_Comment.scss]: 👍

File: client/src/stylesheets/components/_Comment.scss (hunk 17 above)
Comment (hoop71) [reply client/src/stylesheets/components/_Comment.scss]: Seems like it... from before my time.

File: client/src/stylesheets/components/_ToolList.scss (hunk 22 above)
Comment (hoop71) [reply client/src/stylesheets/components/_ToolList.scss]: totally agree but beyond the scope of this PR. We should approach this with media mixins that we can apply everywhere. Will required creating/augmenting _breakpoints.scss and utils/_mixins.scss. 

Opened issue link: https://github.com/codefordenver/Circular/issues/326
